x-pack/plugins/apm/common/agent_name.ts

```diff
@@ -42,20 +42,34 @@ export const AGENT_NAMES: AgentName[] = [
   ...OPEN_TELEMETRY_AGENT_NAMES,
 ];
 
+export const JAVA_AGENT_NAMES: AgentName[] = ['java', 'opentelemetry/java'];
+
+export function isJavaAgentName(
+  agentName?: string
+): agentName is 'java' | 'opentelemetry/java' {
+  return JAVA_AGENT_NAMES.includes(agentName! as AgentName);
+}
```

Comment:
Why do you need to manually cast to `AgentName`?
```suggestion
export function isJavaAgentName(
  agentName?: AgentName
): agentName is 'java' | 'opentelemetry/java' {
  return agentName ? JAVA_AGENT_NAMES.includes(agentName) : false;
}
```

Reply:
> Why do you need to manually cast to `AgentName`?

This makes it possible to narrow the type from `string` to `'java' | 'opentelemetry/java'`.
The input should be string because at this point that's all we know. It's not guaranteed to be a known `AgentName` until parsed.